### PR TITLE
Minimal viable

### PR DESCRIPTION
**Pull request by matthewpeterkort**
code review

### CODE REVIEW

**Comment by bwalsh** [comment]
Two things:
* We should fix the conflicting files with the target branch:
* We should consider squashing commits

**Comment by matthewpeterkort** [comment]
feature/setup branch doesn't need DRSClient.py that can be deleted from feature/setup. feature/setup uses cli.py as the driver file, but this file could be migrated over to main.py in the minimal viable branch, since that seems to be the driver file in the minimal viable branch. File structure is very different between the two branches.

File: drs_downloader/Gen3DRSClient.py [hunk 1]
```diff
@@ -58,74 +58,76 @@ def get_access_url(self, object_id, access_id=None):
 class crdcDRSClient(Gen3DRSClient):
```
Comment (bwalsh) [inline drs_downloader/Gen3DRSClient.py:58]: Re.  these xxxDRSClient classes.  OK to leave in for now?   Let's discuss use case.   Thinking we may want to discard for now and add back in after MVP.

File: drs_downloader/download.py [hunk 2]
```diff
@@ -125,8 +126,13 @@ def hello():
 #     #     exit(1)
```
Comment (bwalsh) [inline drs_downloader/download.py:126]: Remove commented test code

File: tests/test_drs_downloader.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+def test_download_drs():
+    with tempfile.TemporaryDirectory() as dest:
+        download_drs('tests/gen3-data.tsv', dest)
+        files = next(os.walk(dest))[2]
+        assert len(files) == 10
+
+
+def test_get_signed_url():
+    object_url = 'https://development.aced-idp.org/ga4gh/drs/v1/objects/d62af3ff-756d-4e73-b6c6-e0bbd3bae50d'
```
Comment (bwalsh) [inline tests/test_drs_downloader.py:18]: OK for now.  We may need to rethink before release.   (prefer that we don't have to keep a test server up)

**Review by bwalsh — CHANGES_REQUESTED** [review]
Let's discuss after all have reviewed.

File: README.md [hunk 4]
```diff
@@ -18,8 +35,6 @@ Download tool for AnVIL/TDR data identified by DRS URIs and Google
```
Comment (bwalsh) [inline README.md:35]: * Add instructions on how to become a member of our test terra project: https://anvil.terra.bio/#workspaces/anvil-stage-demo/DRS-downloader/data](https://anvil.terra.bio/#workspaces/anvil-stage-demo/DRS-downloader/data

* Add instructions or a link to gcloud setup
* Add instructions to link to the `NHLBI BioData Catalyst Framework Services`.   including `=google` hack

**Review by bwalsh — CHANGES_REQUESTED** [review]

File: drs_downloader/main.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    header = {
+        'authorization': 'Bearer ' + token,
+        'content-type': 'application/json'
+    }
+    async with aiohttp.ClientSession(headers=header) as session:
+        download_urls = await asyncio.gather(*[_get_more(session, uri, url_endpoint) for uri in uris])
+        return download_urls
+
+
+async def _get_more(session: aiohttp.ClientSession, uri: str, endpoint: str) -> DownloadURL:
```
Comment (bwalsh) [inline drs_downloader/main.py:67]: "_get_more" is a bit non-specific.  can we use a more descriptive name?

File: drs_downloader/download.py [hunk 6]
```diff
@@ -101,7 +102,7 @@ class DownloadURL(object):
 
 async def _download(urls: List[DownloadURL], dest: Path):
```
Comment (bwalsh) [inline drs_downloader/download.py:103]: The data class DownloadURL should be renamed to something that describes its contents.
See [DrsObject](https://ga4gh.github.io/data-repository-service-schemas/preview/release/drs-1.0.0/docs/#_drsobject)

Comment (bwalsh) [reply drs_downloader/download.py:103]: Line 86: `assert expected_md5 == actual_md5, f"Actual md5 {actual_md5} does not match expected {expected_md5}"` should not be uncommented

Comment (bwalsh) [reply drs_downloader/download.py:103]: There was a bug in the initial downloader.py


```def parts_generator(size, start=0, part_size=1 * 512 ** 2):
    while size - start > part_size:
        yield start, start + part_size
        start += part_size + 1
    yield start, size
```

**Comment by bwalsh** [comment]
* review and merge PR#4
* start new branch locally "feature/mvp"
* See my comments starting here: https://github.com/anvilproject/drs_downloader/pull/12#pullrequestreview-1173061783
* The new branch should only have a minimum number of files, perhaps only:
```drs_downloader/download.py
drs_downloader/main.py
tests/test_main.py
tests/terra-data.tsv```

**Comment by lbeckman314** [comment]
Closed for #16, all comments above should be resolved in #16.

**Pull request closed without merging**